### DISCUSSION TITLE
UpsertMany function does not work when entity count is 1a

### DISCUSSION THREAD

**Discussion by progressivetr** [mikro-orm/mikro-orm#6055] [General]
### Describe the bug

UpsertMany function does not work when entity count is 1

### Reproduction

UpsertMany function does not work when entity count is 1a

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.3.10

### Node.js version

NodeJS 20 LTS

### Operating system

MacOS

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
Not sure what you mean by that, either way, please provide a complete reproduction so it's clear what you are doing.

https://github.com/mikro-orm/reproduction

**Comment by progressivetr** [comment]
![image](https://github.com/user-attachments/assets/51392dc0-a18a-4f39-bbd8-d994e802c2b9)
![image](https://github.com/user-attachments/assets/7df88c37-e64d-48cd-a2e5-65e9ee070933)

I am using NestJS, and trying to upsert many values in for loop, but when i open debug mode and check queries, in general there is no upsert query. Whats problem here?

**Reply by B4nan** [reply]
if some entity is managed (already loaded in the context), you need to flush to update it, upsert will just call `assign` on it

**Reply by progressivetr** [reply]
How can i disable this feature?

**Reply by B4nan** [reply]
this is how it works, you cant control it right now, you will need to flush or call the upsert on a clear fork. 

```ts
// use on fresh fork
await em.fork().upsertMany();

// or flush afterwards
await em.upsertMany();
await em.flush();
```

but you are not the first one confused by that, i might change this soon, even in next version.

**Reply by B4nan** [reply]
next version will issue the query without the need to flush, see #6197
